### PR TITLE
Caching Implementation for GitHub Actions

### LINKED ISSUE #4150: GitHub Actions Caching Support
#### Nature of issue?

- [ ] Found a bug
- [x] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [ ] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: 
- Web browser and version: 
- Operating System: 
- Steps to reproduce this:

### PR DESCRIPTION
**Pull request by gruselhaus**
Closes #4150

### CODE REVIEW

File: .github/workflows/ci-lint-examples.yml [hunk 1]
```diff
@@ -13,8 +13,14 @@ jobs:
...
         node-version: 12.x
+    - uses: actions/cache@v1
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-node-linting-examples-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-linting-examples-
     - name: Get node modules
-      run: npm ci
+      run: npm i
```
Comment (meiamsome) [inline .github/workflows/ci-lint-examples.yml]: Why change to `npm i`? This means that NPM will install in a configuration that does not match the `package-lock.json` and can lead to some really hard to debug bugs.

Comment (gruselhaus) [reply .github/workflows/ci-lint-examples.yml]: Oh, thanks for the note.

**Comment by limzykenneth** [comment]
I just read up a bit about this and from the [documentation](https://github.com/actions/cache/blob/master/examples.md#node---npm) it isn't recommended to cache `node_modules` and it wouldn't work with `npm ci` anyway since `npm ci` will delete `node_modules` before installing anything.

I don't know whether caching `~/.npm` would be helpful in terms of build times though.

**Comment by outofambit** [comment]
@gruselhaus thanks so much for this pull request. @limzykenneth is right and unfortunately that means we can't use this. (we also don't have a `.npm` folder.)

@gruselhaus i know you helped with the first gh actions PR, so @allcontributors please add @gruselhaus for infra and review.

**Pull request closed without merging**